### PR TITLE
Update libuv commit

### LINKED ISSUE #18818: addprocs works in terminal, fails in Emacs
I have a script that runs fine in the terminal, but fails when I try to run it from Emacs.

I can reproduce this with a file that contains only `addprocs(1)`.

Here's the error output:

```
signal (4): Illegal instruction: 4
while loading /Users/dmatz/Desktop/test.jl, in expression starting on line 1
abort at /usr/local/opt/llvm37-julia/lib/llvm-3.7/lib/libLLVM-3.7.dylib (unknown line)
uv_thread_create at /private/tmp/julia-20160920-99724-ulo33f/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/thread.c:88
init_once at /private/tmp/julia-20160920-99724-ulo33f/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/threadpool.c:173
__pthread_once_handler at /usr/lib/system/libsystem_pthread.dylib (unknown line)
_os_once at /usr/lib/system/libsystem_platform.dylib (unknown line)
pthread_once at /usr/lib/system/libsystem_pthread.dylib (unknown line)
uv_once at /private/tmp/julia-20160920-99724-ulo33f/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/thread.c:239
uv__work_submit at /private/tmp/julia-20160920-99724-ulo33f/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/threadpool.c:184
uv_getaddrinfo at /private/tmp/julia-20160920-99724-ulo33f/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/getaddrinfo.c:186
jl_getaddrinfo at /private/tmp/julia-20160920-99724-ulo33f/src/jl_uv.c:725
getaddrinfo at ./socket.jl:574
jlcall_getaddrinfo_33144 at /usr/local/Cellar/julia/0.5.0/lib/julia/sys.dylib (unknown line)
getaddrinfo at ./socket.jl:589
connect! at ./socket.jl:686
connect at ./stream.jl:959 [inlined]
connect_to_worker at ./managers.jl:483
unknown function (ip: 0x317697039)
jl_call_method_internal at /private/tmp/julia-20160920-99724-ulo33f/src/./julia_internal.h:189 [inlined]
jl_apply_generic at /private/tmp/julia-20160920-99724-ulo33f/src/gf.c:1942
connect at ./managers.jl:425
jlcall_connect_23326 at /usr/local/Cellar/julia/0.5.0/lib/julia/sys.dylib (unknown line)
jl_call_method_internal at /private/tmp/julia-20160920-99724-ulo33f/src/./julia_internal.h:189 [inlined]
jl_apply_generic at /private/tmp/julia-20160920-99724-ulo33f/src/gf.c:1942
create_worker at ./multi.jl:1786
setup_launched_worker at ./multi.jl:1733
unknown function (ip: 0x317692b1d)
jl_call_method_internal at /private/tmp/julia-20160920-99724-ulo33f/src/./julia_internal.h:189 [inlined]
jl_apply_generic at /private/tmp/julia-20160920-99724-ulo33f/src/gf.c:1942
#649 at ./task.jl:360
unknown function (ip: 0x31769273f)
jl_call_method_internal at /private/tmp/julia-20160920-99724-ulo33f/src/./julia_internal.h:189 [inlined]
jl_apply_generic at /private/tmp/julia-20160920-99724-ulo33f/src/gf.c:1942
jl_apply at /private/tmp/julia-20160920-99724-ulo33f/src/./julia.h:1392 [inlined]
start_task at /private/tmp/julia-20160920-99724-ulo33f/src/task.c:253
Allocations: 1273026 (Pool: 1272116; Big: 910); GC: 0

signal (6): Abort trap: 6
while loading /Users/dmatz/Desktop/test.jl, in expression starting on line 1
sigprocmask at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 1273026 (Pool: 1272116; Big: 910); GC: 0
```

After it errors out, there is an orphaned julia process that has 100% CPU usage.

I'm on Mac OS X.  I used homebrew to install Julia.

``` julia
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753* (2016-09-19 18:14 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, ivybridge)
```

### LINKED ISSUE #24169: signal (6): Abort trap:6 on emacs
I'm trying to use Julia and emacs in concert. 

I would really like to use Julia with a nice REPL, so I've been trying to setup julia-mode with julia-repl or Julia-shell or simply ESS, but everything I've tried results in the same error.  

To reproduce:
M-x ansi-term RET
julia> using Requests
julia> import Requests: get, post, put, delete, options
julia> get("http://www.google.dk")

Error:
signal (6): Abort trap: 6
while loading no file, in expression starting on line 0
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 2009569 (Pool: 2008310; Big: 1259); GC: 2
Abort trap: 6

I'm on MacOS High Sierra, Emacs 25.3 and Julia 0.6.0

### PR DESCRIPTION
**Pull request by ararslan**
This includes a fix from upstream libuv that ensures that the thread stack size is set to a multiple of the page size.

Fixes #18818
Fixes #24169

### CODE REVIEW

**Comment by ararslan** [comment]
32-bit Windows failure is unrelated. (Unable to connect to the server to download Cygwin.)

**Comment by ViralBShah** [comment]
What is the difference between the `external dependencies` label and the `upstream` label?

**Comment by ararslan** [comment]
The way I understand it, "upstream" refers to an issue with one of our dependencies that's out of our control, whereas "external dependencies" refers to an issue or PR that addresses something with one of our non-Julia dependencies that doesn't have its own label, e.g. libuv.

**Comment by StefanKarpinski** [comment]
I'm not sure that's a particularly useful distinction. Might be good to fold the two labels into one. Not super important though.

**Comment by ararslan** [comment]
I'm fine with whatever.

**Comment by ararslan** [comment]
Bump @vtjnash

**Comment by ViralBShah** [comment]
Bump @vtjnash for backporting to 0.6.2 (or not)

File: test/socket.jl [hunk 1]
```diff
@@ -414,3 +414,15 @@ end
         @test 0 == ccall(:jl_tcp_reuseport, Int32, (Ptr{Void},), s.handle)
     end
 end
+
+@static if Sys.isapple()
+    @testset "Issues #18818 and #24169" begin
+        ulimit = readchomp(`ulimit -S -s`)
+        try
+            run(`ulimit -S -s 7001`)
```
Comment (vtjnash) [inline test/socket.jl]: this will have no effect

Comment (ararslan) [reply test/socket.jl]: I'm not sure what you mean. Running that locally fails without this change and works with it.

Comment (vtjnash) [reply test/socket.jl]:
```
julia> readchomp(`ulimit -S -s`)
"8192"

julia> readchomp(`ulimit -S -s 7001`)
""

julia> readchomp(`ulimit -S -s`)
"8192"
```

This is a shell-builtin. So you either need to ccall the setrlimit, or spawn this test with a shell that can process both commands.

Comment (ararslan) [reply test/socket.jl]: Okay, I've reimplemented the test with `ccall`. Could you take another look, @vtjnash?

**Review by vtjnash — APPROVED** [review]

**Pull request merged**